### PR TITLE
Implement WPD beamformer

### PR DESCRIPTION
**Pull request by Emrys365**

### CODE REVIEW

File: espnet2/asr/frontend/enh_frontend.py [hunk 1]
```diff
@@ -67,15 +69,15 @@ def forward(
...
         Returns:
-            enhanced spectrum: ComplexTensor, or List[ComplexTensor, ComplexTensor]
-                or predicted magnitude spectrum: torch.Tensor or List[torch.Tensor]
+            enhanced spectrum, or predicted magnitude spectrum:
+                torch.Tensor or List[torch.Tensor]
             output lengths
             predcited masks: OrderedDict[
-                'spk1': List[ComplexTensor(Batch, Frames, Channel, Freq)],
-                'spk2': List[ComplexTensor(Batch, Frames, Channel, Freq)],
+                'spk1': torch.Tensor(Batch, Frames, Channel, Freq),
```
Comment (sw005320) [inline espnet2/asr/frontend/enh_frontend.py:74]: I guess you may not need to import `ComplexTensor` anymore.

**Review by sw005320 — COMMENTED** [review]
- there are several ASR related options (e.g., nlsyms, fbank_pitch) in `wsj0-2mix/enh1/enh.sh`. Do you keep them intentionally? --> correction. Now I found that you also include the ASR evaluation. This is very very good!!!
- I thought that it is good to have dnn_wpe in `espnet2/asr/frontend/nets/dnn_wpe.py`. What is your intention of using `espnet.nets.pytorch_backend.frontends.dnn_wpe`

**Comment by sw005320** [comment]
I did not fully check the detailed lines, but the current design looks good to me.

File: espnet2/asr/frontend/enh_frontend.py (hunk 1 above)
Comment (Emrys365) [reply espnet2/asr/frontend/enh_frontend.py:74]: Yes, I will remove this line:
https://github.com/LiChenda/espnet/blob/2bf70ef7f4cd27aff5213f85e6ec4f755121bee0/espnet2/asr/frontend/enh_frontend.py#L6

**Comment by Emrys365** [comment]
> * there are several ASR related options (e.g., nlsyms, fbank_pitch) in `wsj0-2mix/enh1/enh.sh`. Do you keep them intentionally? --> correction. Now I found that you also include the ASR evaluation. This is very very good!!!

Hmm, I think they are just to be removed later. We are current not including ASR part in enh.sh.

> * I thought that it is good to have dnn_wpe in `espnet2/asr/frontend/nets/dnn_wpe.py`. What is your intention of using `espnet.nets.pytorch_backend.frontends.dnn_wpe`

Previously I was trying to design a multi-output WPE module for the multi-speaker case. After discussing with Aswin, I feel that may not be necessary for now. So I simply use the existing `espnet.nets.pytorch_backend.frontends.dnn_wpe`. If we want to make some changes to that module, I will create a new one in `espnet2/asr/frontend/nets/dnn_wpe.py`.

**Comment by sw005320** [comment]
> > * there are several ASR related options (e.g., nlsyms, fbank_pitch) in `wsj0-2mix/enh1/enh.sh`. Do you keep them intentionally? --> correction. Now I found that you also include the ASR evaluation. This is very very good!!!
> 
> Hmm, I think they are just to be removed later. We are current not including ASR part in enh.sh.
> 
> > * I thought that it is good to have dnn_wpe in `espnet2/asr/frontend/nets/dnn_wpe.py`. What is your intention of using `espnet.nets.pytorch_backend.frontends.dnn_wpe`
> 
> Previously I was trying to design a multi-output WPE module for the multi-speaker case. After discussing with Aswin, I feel that may not be necessary for now. So I simply use the existing `espnet.nets.pytorch_backend.frontends.dnn_wpe`. If we want to make some changes to that module, I will create a new one in `espnet2/asr/frontend/nets/dnn_wpe.py`.

I see.
Thanks!

**Pull request merged**